src/messaging/ExchangeContext.cpp

```diff
@@ -573,5 +579,20 @@ ExchangeMessageDispatch & ExchangeContext::GetMessageDispatch(bool isEphemeralEx
     return ApplicationExchangeDispatch::Instance();
 }
 
+void ExchangeContext::AbortAllOtherCommunicationOnFabric()
```

Comment:
ExchangeMgr would be a better bet honestly barring the above suggestion.

Reply:
So the reason this is on the ExchangeContext is because that's the entity that has the one special SessionHolder that we want to exempt from eviction....

Reply:
See [this](https://github.com/project-chip/connectedhomeip/pull/19780#discussion_r902976774) comment.

Reply:
> that's the entity that has the one special SessionHolder that we want to exempt from eviction....

Can't you not extract the `mSession` from that through a getter and implement all of this same bits of logic in `ExchangeManager`?

Reply:
We _could_ extract it through a getter.  Then we'd need to keep the public setter for the "tear down the session when this exchange goes away" bit (which becomes non-public in this PR).

I can go either way, but I was trying to minimize the number of footgun APIs.  At least this one I am adding corresponds to a clear spec-defined operation...